lib/ok.ex

```diff
@@ -150,32 +150,52 @@ defmodule OK do
   end
 
   @doc """
-  Combine multiple functions that may return an error.
-
-  `OK.with/1 is a strict version of the native elixir with special form.
-  It is stricter because it will only handle result tuples.
-  This however allows for more terse expressions.
+  Composes multiple functions similar to Elixir's native `with` construct.
+  
+  `OK.with/1` enables more terse and readable expressions however, eliminating
+  noise and regaining precious horizontal real estate in the process. It does 
+  this by extracting result tuples when using the `<-` operator.
+  
+  For example, in native Elixir, you could write a `with` statement as follows:
+  
+  ```elixir
+  with {:ok, a} <- safe_div(8,2),
+       {:ok, b} <- safe_div(a,2),
+       do: {:ok, b}
+  ```
+  
+  With `OK.with/1`, this can be rewritten as follows:
+  ```elixir
+  OK.with do
+    a <- safe_div(8,2)
+    b <- safe_div(a,2)
+    {:ok, b}
+  end
+  ```
 
-  When combining funcations that may fail the result pipe operator is inflexible in several areas.
+  ### Compared to `~>>`
```

Comment:
I don't think we need a compared to section I probably wrote it like that just cz of the order I developed it in. perhaps just limit the docs to saying what it does rather than what other things doo